specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-03-01/storagecache.json

```diff
@@ -0,0 +1,1776 @@
+            }
+          }
+        }
+      ]
+    },
+    "StorageTargetResource": {
+      "description": "Resource used by a Cache.",
+      "properties": {
+        "name": {
+          "description": "Name of the Storage Target.",
+          "readOnly": true,
+          "$ref": "#/definitions/ResourceName"
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID of the Storage Target."
+        },
+        "type": {
+          "description": "Type of the Storage Target; Microsoft.StorageCache/Cache/StorageTarget",
+          "readOnly": true,
+          "type": "string"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "StorageTargetProperties": {
+      "type": "object",
+      "description": "Properties of the Storage Target.",
+      "discriminator": "targetBaseType",
+      "properties": {
+        "targetBaseType": {
+          "description": "Type of the Storage Target.",
+          "$ref": "#/definitions/StorageTargetType"
+        },
+        "junctions": {
+          "description": "List of Cache namespace junctions to target for namespace associations.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NamespaceJunction"
+          }
+        },
+        "targetType": {
+          "description": "Type of the Storage Target.",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "ARM provisioning state, see https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property",
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Cancelled",
+            "Creating",
+            "Deleting",
+            "Updating"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningStateType",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "nfs3": {
+          "description": "Properties when targetType is nfs3.",
+          "$ref": "#/definitions/Nfs3Target"
+        },
+        "clfs": {
+          "description": "Properties when targetType is clfs.",
+          "$ref": "#/definitions/ClfsTarget"
+        },
+        "unknown": {
+          "description": "Properties when targetType is unknown.",
+          "$ref": "#/definitions/UnknownTarget"
+        }
+      },
+      "required": [
+        "targetBaseType"
+      ]
+    },
+    "StorageTargetType": {
+      "type": "string",
+      "description": "Type of the Storage Target.",
+      "enum": [
+        "nfs3",
+        "clfs",
+        "unknown"
+      ],
+      "x-ms-enum": {
+        "name": "StorageTargetType",
+        "modelAsString": true
+      }
+    },
+    "Nfs3TargetProperties": {
+      "description": "An NFSv3 mount point for use as a Storage Target.",
+      "type": "object",
+      "x-ms-discriminator-value": "nfs3",
```

Comment:
>"x-ms-discriminator-value": "nfs3", [](start = 6, length = 35)

I don't see benefit of using discriminator here, since these are explicitly referenced under the field/key 
nfs3": 
clfs:
unknown:

Instead , the StorageTargetProperties itself should be polymorphic based discriminator

Reply:
Nfs3TargetProperties, ClfsTargetProperties and UnknownTargetProperties are actually derived from StorageTargetProperties. I think it is related to your next comment.